apps/judicial-system/backend/src/app/modules/case/limitedAccessCase.service.ts

```diff
@@ -540,23 +546,71 @@ export class LimitedAccessCaseService {
             { reason },
           ),
         )
+    })
+
+    await Promise.all(fileFetchPromises)
+
+    if (isRequestCase(theCase.type)) {
+      filesToZip.push(
+        {
+          data: await this.pdfService.getRequestPdf(theCase),
+          name: 'Krafa.pdf',
+        },
+        {
+          data: await this.pdfService.getCourtRecordPdf(theCase, user),
+          name: 'Þingbók.pdf',
+        },
+        {
+          data: await this.pdfService.getRulingPdf(theCase),
+          name: 'Úrskurður.pdf',
+        },
+      )
     }
 
-    filesToZip.push(
-      {
-        data: await this.pdfService.getRequestPdf(theCase),
-        name: 'krafa.pdf',
-      },
-      {
-        data: await this.pdfService.getCourtRecordPdf(theCase, user),
-        name: 'þingbok.pdf',
-      },
-      {
-        data: await this.pdfService.getRulingPdf(theCase),
-        name: 'urskurður.pdf',
-      },
-    )
+    if (isIndictmentCase(theCase.type)) {
+      const defendant = theCase.defendants && theCase.defendants[0]
+      const subpoena =
+        defendant && defendant.subpoenas && defendant.subpoenas[0]
+
+      filesToZip.push(
+        {
+          data: await this.pdfService.getIndictmentPdf(theCase),
+          name: 'Ákæra.pdf',
+        },
+        {
+          data: await this.pdfService.getCaseFilesRecordPdf(
+            theCase,
+            theCase.policeCaseNumbers[0],
```

Comment:
There is a separate case files record for each police csae number, so we need to return them all. Also not that each case files record is identified by the police case number, not the court case number. Please make the necessary modifications.